### PR TITLE
Fix: Token Price uses correct reader

### PR DESCRIPTION
**Pull request by asoliman92**

### CODE REVIEW

File: pkg/reader/price_reader.go [hunk 1]
```diff
@@ -184,7 +190,7 @@ func (pr *priceReader) getFeedDecimals(
 ) (uint8, error) {
 	var decimals uint8
 	if err :=
-		pr.feedChainReader.GetLatestValue(
+		pr.feedChainReader().GetLatestValue(
```
Comment (makramkd) [inline pkg/reader/price_reader.go:193]: Can be nil, is this covered by the nil check above?

File: pkg/reader/price_reader.go [hunk 2]
```diff
@@ -106,7 +112,7 @@ func (pr *priceReader) GetFeeQuoterTokenUpdates(
 	// MethodNameFeeQuoterGetTokenPrices returns an empty update with
 	// a timestamp and price of 0 if the token is not found
 	if err :=
-		pr.feedChainReader.GetLatestValue(
+		pr.chainReaders[chain].GetLatestValue(
```
Comment (makramkd) [inline pkg/reader/price_reader.go:115]: This can be nil, need to do a nil check somewhere

File: pkg/reader/price_reader.go (hunk 1 above)
Comment (asoliman92) [reply pkg/reader/price_reader.go:193]: Yes, it's covered by the nil check above, doing extra won't hurt though.

File: commit/factory.go [hunk 3]
```diff
@@ -159,7 +159,7 @@ func (p *PluginFactory) NewReportingPlugin(ctx context.Context, config ocr3types
 	)
 
 	// The node supports the chain that the token prices are on.
-	tokenPricesCr, ok := p.contractReaders[offchainConfig.PriceFeedChainSelector]
+	_, ok := readers[offchainConfig.PriceFeedChainSelector]
```
Comment (winder) [inline commit/factory.go:162]: I'm surprised that you need to switch from `p.contractReaders` to `readers`. Is that really required?

We can move the facade mapping to `NewPluginFactory` and avoid using the actual ContractReader type outside of the constructor.

Comment (asoliman92) [reply commit/factory.go:162]: I don't "need" to, it felt more consistent to use the same `facade` everywhere though.

**Review by winder — APPROVED** [review]

**Review by mateusz-sekara — COMMENTED** [review]
Can we have a test covering this fix? I mean failing before applying your fixes

**Pull request merged**